types/api/config.d.ts

```diff
@@ -46,6 +46,7 @@ interface ClientConfig {
     EnableBanner: string;
     EnableBotAccountCreation: string;
     EnableChannelViewedMessages: string;
+    EnableClientMetrics: string;
```

Comment:
I have mixed feelings, since we have to cater for older server versions, and the webapp doesn't. So types that are still relevant for mobile could be removed from `@mattermost/types`.

Reply:
I do have mix feelings as well, we tried to do this in the past and it went a drift really fast and in this case will be specially true as that package does not really consider backwards compatibility with server versions, thus we could expect that a prop is available when it does not even exist.

That said, `EnableClientMetrics` should be optional

Reply:
We might accidentally be keeping backwards compatibility in mind given we forget to delete old fields from it all the time, but I see what you mean there. I wonder if plugins run into the same issue since they also use `@mattermost/types` and will similarly be targetting a range of MM server versions.

Regardless, we can definitely leave that as-is here since that's far outside the scope of this PR